src/main/java11/org/kohsuke/github/extras/HttpClientGitHubConnector.java

```diff
@@ -104,4 +159,49 @@ public void close() throws IOException {
             IOUtils.closeQuietly(response.body());
         }
     }
+
+    private static URI getRedirectedUri(URI originalUri, HttpHeaders headers) throws IOException {
+        URI redirectedURI;
+        redirectedURI = headers.firstValue("Location")
+                .map(URI::create)
+                .orElseThrow(() -> new IOException("Invalid redirection"));
+
+        // redirect could be relative to original URL, but if not
+        // then redirect is used.
+        redirectedURI = originalUri.resolve(redirectedURI);
+        return redirectedURI;
+    }
+
+    // This implements the exact same rules as the ones applied in RedirectFilter
```

Comment:
`RedirectFilter`?  What project is this from?

Reply:
Ah yes, sorry, it's the JDK class. I should have added the package.

Reply:
It's `jdk.internal.net.http.RedirectFilter`.

Reply:
```suggestion
    // This implements the exact same rules as the ones applied in jdk.internal.net.http.RedirectFilter
```